### PR TITLE
New Keyboard: Think 6.5

### PR DESCRIPTION
**Pull request by mechmerlin**
## Description

New Keyboard from Gray Studio, the Think6.5 run by Airpotter and Oldcat. 

Thank you to sarcasm/bli08 for testing this code on his hotswap Think6.5. 

Thanks to Moridin and Cygnus for providing the kbfirmware files to me. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by mechmerlin** [comment]
On hold, discovered new information

File: keyboards/gray_studio/think65/hotswap/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/* Copyright 2019 REPLACE_WITH_YOUR_NAME
```
Comment (fauxpark) [inline keyboards/gray_studio/think65/hotswap/keymaps/default/keymap.c]:
```suggestion
/* Copyright 2019 MechMerlin
```

File: keyboards/gray_studio/think65/hotswap/rules.mk [hunk 2]
```diff
@@ -0,0 +1,38 @@
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   atmega32a    bootloadHID
```
Comment (fauxpark) [inline keyboards/gray_studio/think65/hotswap/rules.mk]:
```suggestion
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
```

File: keyboards/gray_studio/think65/hotswap/rules.mk [hunk 3]
```diff
@@ -0,0 +1,38 @@
+MCU = atmega32u4
```
Comment (fauxpark) [inline keyboards/gray_studio/think65/hotswap/rules.mk]:
```suggestion
# MCU name
MCU = atmega32u4
```

File: keyboards/gray_studio/think65/solder/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/* Copyright 2019 REPLACE_WITH_YOUR_NAME
```
Comment (fauxpark) [inline keyboards/gray_studio/think65/solder/keymaps/default/keymap.c]:
```suggestion
/* Copyright 2019 MechMerlin
```

File: keyboards/gray_studio/think65/solder/rules.mk [hunk 5]
```diff
@@ -0,0 +1,38 @@
+MCU = atmega32u4
```
Comment (fauxpark) [inline keyboards/gray_studio/think65/solder/rules.mk]:
```suggestion
# MCU name
MCU = atmega32u4
```

File: keyboards/gray_studio/think65/solder/rules.mk [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   atmega32a    bootloadHID
```
Comment (fauxpark) [inline keyboards/gray_studio/think65/solder/rules.mk]:
```suggestion
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
```

**Pull request merged**